src/content/1.7/modules/sample_modules/order_view_page_new_hooks/view-order-hooks-signature-widget.md

```diff
@@ -0,0 +1,448 @@
+     */
+    private $filename;
+
+    /**
+     * @return int|null
+     */
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int $id
+     *
+     * @return self
+     */
+    public function setId(int $id): self
+    {
+        $this->id = $id;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getFilename(): string
+    {
+        return $this->filename;
+    }
+
+    /**
+     * @param string $filename
+     *
+     * @return self
+     */
+    public function setFilename(string $filename): self
+    {
+        $this->filename = $filename;
+
+        return $this;
+    }
+
+    /**
+     * @return int
+     */
+    public function getOrderId(): int
+    {
+        return $this->orderId;
+    }
+
+}
+
+```
+
+Let's put our signature picture, `john_doe.png` inside `/signatures/` folder.
+
+ {{< figure src="../../img/view-order-hooks-demo/john_doe.png" title="Signature" >}}
+
+Let's create Order Repository and data structures for interacting with Orders data.
+Data Transfer Object (https://en.wikipedia.org/wiki/Data_transfer_object)
+`Order.php` in `src/DTO` to carry data between processes. The main benefit is that it reduces
+the amount of data that needs to be sent inside application and encapsulates parameters for
+method calls (This can be useful if a method takes more than 4 or 5 parameters.). Also it is more 
+convenient than using associative array where you need to remember it's indexes to get the 
+certain values (no autocomplete is provided).
+
+```php
+<?php
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\DTO;
+
+use DateTimeImmutable;
+
+final class Order
+{
+    /**
+     * @var int
+     */
+    private $orderId;
+
+    /**
+     * @var string
+     */
+    private $reference;
+
+    /**
+     * @var int
+     */
+    private $orderStateId;
+
+    /**
+     * @var DateTimeImmutable
+     */
+    private $orderDate;
+
+    public function __construct(int $orderId, string $reference, int $orderStateId, DateTimeImmutable $orderDate)
+    {
+        $this->orderId = $orderId;
+        $this->reference = $reference;
+        $this->orderStateId = $orderStateId;
+        $this->orderDate = $orderDate;
+    }
+
+    public function getOrderId(): int
+    {
+        return $this->orderId;
+    }
+
+    public function getReference(): string
+    {
+        return $this->reference;
+    }
+}
+```
+
+Orders collection `Orders.php` in `src/Collection`:
+
+```php
+<?php
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Collection;
+
+use PrestaShop\Module\DemoViewOrderHooks\DTO\Order;
+use PrestaShop\PrestaShop\Core\Data\AbstractTypedCollection;
+
+final class Orders extends AbstractTypedCollection
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function getType()
+    {
+        return Order::class;
+    }
+}
+```
+
+`OrderRepository` in `src\Repository`:
```

Comment:
We should explain the difference between this "legacy" repository, compare to the doctrine repository `OrderSignatureRepository` something like:
```
{{% notice note %}}
Unlike the previous `OrderSignatureRepository` this repository uses legacy PrestaShop classes, it performs the request using the `Db` class. We have to do this because core objects from PrestaShop do not use Doctrine entities, so we can't use a Doctrine repository to manage them.
{{% /notice %}}
```

Reply:
love it! :heart_decoration: